src/modules/disk_io.cpp

```diff
@@ -0,0 +1,157 @@
+#include <fstream>
+#include <iomanip>
+#include <istream>
+#include <vector>
+#include <regex>
+#include <sys/types.h>
+#include <dirent.h>
+
+#include "drawtypes/label.hpp"
+#include "drawtypes/progressbar.hpp"
+#include "modules/disk_io.hpp"
+#include "utils/math.hpp"
+
+#include "modules/meta/base.inl"
+
+POLYBAR_NS
+
+namespace modules {
+  template class module<disk_io_module>;
+
+  disk_io_module::disk_io_module(const bar_settings& bar, string name_) : timer_module<disk_io_module>(bar, move(name_)) {
+    m_interval = m_conf.get<decltype(m_interval)>(name(), "interval", 1s);
+
+    m_formatter->add(DEFAULT_FORMAT, TAG_LABEL, {TAG_LABEL}/*, TAG_BAR_READ, TAG_BAR_WRITE}*/);
+    if (m_formatter->has(TAG_LABEL)) {
+      m_label = load_optional_label(m_conf, name(), TAG_LABEL, "%speed_read% Mb/s %speed_write% Mb/s");
+    }
+    m_disk_names = _get_disk_names();
+    for (auto name : m_disk_names) {
+      std::cout << "name: " << name << std::endl;
+    }
+}
+
+std::vector<std::string> disk_io_module::_get_disk_names(void) {
+  DIR *dp;
+  struct dirent *dirp;
+  std::vector<std::string> names;
+
+  if (!(dp = opendir("/sys/block/")))
+  {
+    std::cout << "Can't open /sys/block!!" << std::endl;
+    return names;
+  }
+  while ((dirp = readdir(dp))) {
+    if (dirp->d_name[0] != '.') {
+      names.push_back(std::string(dirp->d_name));
+    }
+  }
+  closedir(dp);
+  return names;
+}
+
+std::pair<unsigned long long, unsigned long long> disk_io_module::_get_disk_read_write(std::string disk_name) {
+  unsigned long long sec_read_total_new{0ULL};
+  unsigned long long sec_write_total_new{0ULL};
+  try {
+    std::ifstream disk_io_info("/sys/block/" + disk_name + "/stat");
+
+    std::string line;
+    std::getline(disk_io_info, line);
+
+    static const std::regex re{"\\s+"};
+    std::vector<std::string> items{
+      std::sregex_token_iterator(line.begin(), line.end(), re, -1),
+        std::sregex_token_iterator()
+    };
+
+    std::vector<std::string> filtered_items;
+
+    for (const auto &item : items) {
+      if (item.length() > 0)
+        filtered_items.push_back(item);
+    }
+
+    sec_read_total_new = std::strtoull(filtered_items[READ_SECTORS_OFFSET].c_str(), nullptr, 10);
+    sec_write_total_new = std::strtoull(filtered_items[WRITE_SECTORS_OFFSET].c_str(), nullptr, 10);
+  } catch (const std::exception& err) {
+    m_log.err("Failed to read memory values (what: %s)", err.what());
+  }
+  return (std::make_pair(sec_read_total_new, sec_write_total_new));
+}
+
+float disk_io_module::_get_time_delta(void) {
+  std::chrono::milliseconds now = std::chrono::duration_cast<std::chrono::milliseconds>(
+      std::chrono::system_clock::now().time_since_epoch());
+
+  std::chrono::duration<float> ms_delta = now - m_ms;
+  float time_delta = ms_delta.count();
+  m_ms = now;
+  return time_delta;
+}
+
+void disk_io_module::_calculate_disk_io_speeds(std::string disk_name)
```

Comment:
You should use `string_util::filesize()` if you wanted (kilo/mega)bytes/s, or create another `string_util` function to calculate it in (kilo/mega)bits/s.